src/main/java/org/jenkinsci/plugins/fodupload/steps/FortifyStaticAssessment.java

```diff
@@ -183,9 +199,13 @@ public void perform(Run<?, ?> build, FilePath workspace, Launcher launcher, Task
                 entitlementPreference,
                 srcLocation,
                 remediationScanPreferenceType,
-                inProgressScanActionType);
+                inProgressScanActionType,
+                inProgressBuildResultType);
 
         commonBuildStep.perform(build, workspace, launcher, listener);
+        CrossBuildAction crossBuildAction = build.getAction(CrossBuildAction.class);
+        crossBuildAction.setPreviousStepBuildResult(build.getResult());
+        try{build.save();} catch(IOException ex){log.println("Error saving settings. Error message: " + ex.toString());}
```

Comment:
If we can't save the build here with the new CrossBuildAction, will the new warning option not work properly?  Should we re-throw the exception here if not?

Reply:
Their ability to save is based off of the credentials they use. If the save fails, the later in progress scan checks will still end the polling.